reconcile/openshift_resources_base.py

```diff
@@ -600,6 +606,15 @@ def fetch_provider_vault_secret(
         raise FetchResourceError(str(e))
 
 
+# check to ensure that all of the keys are valid by looking to see if there are any white space issues. If any issues are uncovered, an exception will be raised.
+def assert_valid_secret_keys(items):
```

Comment:
A better check would be to abide by the spec. https://kubernetes.io/docs/concepts/configuration/secret/#restriction-names-data 

> The keys of data and stringData must consist of alphanumeric characters, -, _ or .

I also suggest updating the error message so it's more user friendly. The error message should point out vault path and key, in addition to the spec requirements. Often times I see our tenants struggling with the lack of good error messaging, so small things like this definitely adds up.

Reply:
that's a good point @bhushanthakur93